### DISCUSSION TITLE
s3 sync output logging

### DISCUSSION THREAD

**Discussion by arthur-guru** [aws/aws-cli#7509] [General]
Hello

aws cli s3 sync is very handy and powerful however the logging is a little terse and unconventional.
E.g. 
upload: src_object_path_1 to dst_object_path_1
upload: src_object_path_2 to dst_object_path_2

Looking at the s3 source code it appears that details of the objects that qualify to be synced across are already available at the time of the PutObject request and the subsequent output prompt above.

The aws cli option  --output json doesn't produce anything when using the s3 sync option which is odd.

Therefore could an extra option be added to the aws s3 sync output to also indicate the bytes transferred.
E.g. 
upload: src_object_path_1 to dst_object_path_1 (41234 bytes)
upload: src_object_path_2 to dst_object_path_2 (798 bytes)

This will then allow me to log both object movement and bytes in my reports.

Thank you, Arthur G.

**Comment by tim-finnigan** [comment]
Hi @arthur-guru thanks for reaching out and apologies for not following up here sooner. Our focus is more on discussions in the [Q&A](https://github.com/aws/aws-cli/discussions/categories/q-a) category so this one was not on our radar. 

It sounds like what you're describing is a feature request for logging file size during s3 upload. For new feature requests we recommend filling out this template: https://github.com/aws/aws-cli/issues/new?assignees=&labels=feature-request%2Cneeds-triage&template=feature-request.yml&title=%28short+issue+description%29

Please fill out the information there and we can follow up on that issue. Thanks!

**Comment by arthur-guru** [comment]
Hi Tim,

Many thanks for the guidance, I've since lodged the following feature request:
https://github.com/aws/aws-cli/issues/7536

Thanks, Arthur.
